flocker/control/test/test_schemas.py

```diff
@@ -475,97 +475,114 @@
-
-        # too short string for dataset_id
-        {u"primary": u"10.0.0.1", u"dataset_id": u"x" * 35},
-
-        # too long string for dataset_id
-        {u"primary": u"10.0.0.1", u"dataset_id": u"x" * 37},
-
-        # wrong type for metadata
-        {u"primary": u"10.0.0.1", u"metadata": 10},
-
-        # wrong type for value in metadata
-        {u"primary": u"10.0.0.1", u"metadata": {u"foo": 10}},
-
-        # too-long string property name in metadata
-        {u"primary": u"10.0.0.1", u"metadata": {u"x" * 257: u"10"}},
-
-        # too-long string property value in metadata
-        {u"primary": u"10.0.0.1", u"metadata": {u"foo": u"x" * 257}},
-
-        # too many metadata properties
-        {u"primary": u"10.0.0.1",
-         u"metadata":
-             dict.fromkeys((unicode(i) for i in range(257)), u"value")},
-
-        # wrong type for maximum size
-        {u"primary": u"10.0.0.1", u"maximum_size": u"123"},
-
-        # too-small value for maximum size
-        {u"primary": u"10.0.0.1", u"maximum_size": 123},
+    failing_instances=CONFIGURATION_DATASET_FAILING_INSTANCES,
+    passing_instances=CONFIGURATION_DATASET_PASSING_INSTANCES,
+)
 
-        # missing primary
-        {u"metadata": {},
-         u"maximum_size": 1024 * 1024 * 1024,
-         u"dataset_id": u"x" * 36},
-
-        # wrong type for primary
-        {u"primary": 10,
-         u"metadata": {},
-         u"maximum_size": 1024 * 1024 * 1024,
-         u"dataset_id": u"x" * 36},
-
-        # non-IPv4-address for primary
-        {u"primary": u"10.0.0.257",
-         u"metadata": {},
-         u"maximum_size": 1024 * 1024 * 1024,
-         u"dataset_id": u"x" * 36},
-        {u"primary": u"example.com",
-         u"metadata": {},
-         u"maximum_size": 1024 * 1024 * 1024,
-         u"dataset_id": u"x" * 36},
-
-        # wrong type for deleted
-        {u"primary": u"10.0.0.1",
-         u"deleted": u"hello"},
-    ],
 
-    passing_instances=[
-        # everything optional except primary
-        {u"primary": u"10.0.0.1"},
-
-        # metadata is an object with a handful of short string key/values
-        {u"primary": u"10.0.0.1",
-         u"metadata":
-             dict.fromkeys((unicode(i) for i in range(16)), u"x" * 256)},
-
-        # maximum_size is an integer of at least 64MiB
-        {u"primary": u"10.0.0.1", u"maximum_size": 1024 * 1024 * 64},
-
-        # dataset_id is a string of 36 characters
-        {u"primary": u"10.0.0.1", u"dataset_id": u"x" * 36},
-
-        # deleted is a boolean
-        {u"primary": u"10.0.0.1", u"deleted": False},
-
-        # All of them can be combined.
-        {u"primary": u"10.0.0.1",
-         u"metadata":
-             dict.fromkeys((unicode(i) for i in range(16)), u"x" * 256),
-         u"maximum_size": 1024 * 1024 * 64,
-         u"dataset_id": u"x" * 36,
-         u"deleted": True},
-    ]
+ConfigurationDatasetsNewSchemaTests = build_schema_test(
+    name="ConfigurationDatasetsPrimarySchemaTests",
```

Comment:
Switch the name here to something better (without "New") and make the identifier and the `name` agree.